lib/src/utils.dart

```diff
@@ -12,6 +14,47 @@ T min<T extends Comparable>(T obj1, T obj2) =>
 T max<T extends Comparable>(T obj1, T obj2) =>
     obj1.compareTo(obj2) > 0 ? obj1 : obj2;
 
+/// Returns whether all elements of [iter] are the same value, according to
+/// `==`.
+///
+/// Assumes [iter] doesn't contain any `null` values.
+bool isAllTheSame(Iterable<Object> iter) {
+  Object lastValue;
+  for (var value in iter) {
+    if (lastValue == null) {
+      lastValue = value;
+    } else if (value != lastValue) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/// Returns whether [span] covers multiple lines.
+bool isMultiline(SourceSpan span) => span.start.line != span.end.line;
+
+/// Sets the first `null` element of [list] to [element].
+void setFirstNull<E>(List<E> list, E element) {
+  for (var i = 0; i < list.length; i++) {
+    if (list[i] == null) {
+      list[i] = element;
+      return;
+    }
+  }
+  throw ArgumentError("$list contains no null elements.");
+}
+
+/// Sets the element of [list] that currently contains [element] to `null`.
+void setToNull<E>(List<E> list, E element) {
+  for (var i = 0; i < list.length; i++) {
```

Comment:
Can we use `indexOf`?

Also, how about `replaceWith` instead of `setTo`?

```dart
void replaceWithNull<E>(List<E> list, E element) {
  var index = list.indexOf(element);
  if (index < 0) throw ArgumentError(...);
  list[index] = null;
}
```

Reply:
Done.